src/Compilers/Core/Portable/Text/SourceText.cs

```diff
@@ -631,6 +636,7 @@ ImmutableArray<byte> computeContentHash()
                 var hash = s_contentHashPool.Allocate();
                 var charBuffer = s_charArrayPool.Allocate();
                 Debug.Assert(charBuffer.Length == CharBufferSize);
+                var isBigEndian = !BitConverter.IsLittleEndian;
```

Comment:
If you wrote `if (!BitConverter.IsLittleEndian)`, the JIT will be able to dead-code eliminate the whole block on little endian. I'm not 100% sure its optimizations will track this local through to the usage and achieve the same effect. It might, I'm just not confident.

Reply:
got it.